File: src/luckyNumber.cpp
#include<bits/stdc++.h>
using namespace std;

bool luckyNumber(int n){

    bool count[10];
    for(int i = 0;i<10;i++)
        count[i] = 0;

    while(n > 0){
        int dig = n%10;
        if(count[dig])
            return false;
        n /= 10;
    }
    return true;

}

int main(){

    int n;
    cout<<"Enter a number :";
    cin>>n;

    bool ans = luckyNumber(n);
    if(ans){
        cout<<"It is a lucky number";
    }
    else{
        cout<<"It is not a lucky number";
    }

    return 0;
}

File: src/luckyNumber.py
#!/usr/bin/env python3
# -*- coding: utf-8 -*-
"""
Created on Fri May 31 10:38:20 2019

@author: agupta47
"""

""" Description (optional)
Program to check if a given number is Lucky (all digits are different)
Input: n = 983
Output: true
All digits are different

Input: n = 9838
Output: false
8 appears twice
"""

n = int(input("Enter a number : "))
dic = set()

def lucky(n):
    while(n>0):
        tmp = n%10
        if tmp in dic:
            print("False")
            print("{tmp} appears twice")
            return 
        else:
            dic.add(int(tmp))
        n = n/10
    print("True")
    print("All digits are different")
    return

lucky(n)
    